extensions/intellij/src/main/kotlin/dev/tigr/melody/plugin/Melody.bnf

```diff
@@ -0,0 +1,101 @@
+{
+  parserClass="dev.tigr.melody.plugin.parser.MelodyParser"
+
+  extends="com.intellij.extapi.psi.ASTWrapperPsiElement"
+
+  psiClassPrefix="Melody"
+  psiImplClassSuffix="Impl"
+  psiPackage="dev.tigr.melody.plugin.psi"
+  psiImplPackage="dev.tigr.melody.plugin.psi.impl"
+
+  elementTypeHolderClass="dev.tigr.melody.plugin.psi.MelodyTypes"
+  elementTypeClass="dev.tigr.melody.plugin.psi.MelodyElementType"
+  tokenTypeClass="dev.tigr.melody.plugin.psi.MelodyTokenType"
+
+  tokens=[
+    // keywords
+    of="of"
+    to="to"
+    capture="capture"
+    some="some"
+    match="match"
+    over="over"
+    option="option"
+    not="not"
+    either="either"
+    any="any"
+    ahead="ahead"
+    behind="behind"
+    let="let"
+    lazy="lazy"
+
+    // symbols
+    start="<start>"
+    end="<end>"
+    char="<char>"
+    whitespaceliteral="<whitespace>"
+    space="<space>"
+    newline="<newline>"
+    tab="<tab>"
+    return="<return>"
+    feed="<feed>"
+    null="<null>"
+    digit="<digit>"
+    vertical="<vertical>"
+    word="<word>"
+    alphabet="<alphabetic>"
+    alphanumeric="<alphanumeric>"
+    boundary="<boundary>"
+    backspace="<backspace>"
+
+    // numbers
+    number="regexp:[0-9]+"
+
+    // comments
+    comment="regexp://.*|/\*!(\*/)*\*/|/\*\*/"
+
+    // strings
+    string = "regexp:\"[^\"]*\"|'[^']*'|`[^`]*`"
+
+    // other patterns
+    character = "regexp:\S"
```

Comment:
character literals (for negative classes) only support these at the moment: `('a'..'z'|'A'..'Z'| '0'..'9' | "_" | "\\" | "@" | "*" | "$" | "#" | "&" | "^" | "!" | "%")`

Reply:
Ok, I just pushed those changes!